ddtrace/internal/datastreams/processor.py

```diff
@@ -312,10 +312,15 @@ def decode_pathway(self, data):
             return self.new_pathway()
 
     def decode_pathway_b64(self, data):
-        # type: (Optional[str]) -> DataStreamsCtx
+        # type: (Optional[Union[str, bytes]]) -> DataStreamsCtx
         if not data:
             return self.new_pathway()
-        binary_pathway = data.encode("utf-8")
+
+        if isinstance(data, str):
+            binary_pathway = data.encode("utf-8")
+        else:
+            binary_pathway = data
```

Comment:
Is this a bug fix? Do we need a release note here

Reply:
Not a bug fix. This stems from the fact that some messaging services (Kafka) convert the headers from strings to bytes. However, we weren't using this function to decode kafka headers since kafka was still using the deprecated encoding for DSM, we were using `decode_pathway` which accepts bytes.